From: yangsong8 <yangsong8@xiaomi.com>
Date: Tue, 6 May 2025 20:59:34 +0800
Subject: [PATCH] drivers/usbdev/cdcacm.c: fix runtime error memcpy para is
 null

usbdev/cdcacm.c:2974:3: runtime error: null pointer passed as argument 2, which is declared to never be null
    #0 0x44716307 in cdcuart_dmasend usbdev/cdcacm.c:2974

Signed-off-by: yangsong8 <yangsong8@xiaomi.com>

diff --git a/drivers/usbdev/cdcacm.c b/drivers/usbdev/cdcacm.c
index 45862ead98..9076547b28 100644
--- a/drivers/usbdev/cdcacm.c
+++ b/drivers/usbdev/cdcacm.c
@@ -2982,8 +2982,12 @@ static void cdcuart_dmasend(FAR struct uart_dev_s *dev)
   req->len = nbytes;
 
   nbytes = MIN(reqlen - nbytes, xfer->nlength);
-  memcpy(req->buf + req->len, xfer->nbuffer, nbytes);
-  req->len += nbytes;
+  if (nbytes)
+    {
+      memcpy(req->buf + req->len, xfer->nbuffer, nbytes);
+      req->len += nbytes;
+    }
+
   xfer->nbytes = req->len;
 
   uart_xmitchars_done(dev);
